### ISSUE TITLE
Compose Service Up Tasks Keep Getting Stopped

### ISSUE THREAD

**Issue by migdall** [aws/amazon-ecs-cli#423]
We are deploying our Docker images through ECS and creating the cluster is working just fine with the following command.

`ecs-cli up --force --keypair <redacted> --capability-iam --size 1 --instance-type t2.large`

Following that we already have a repository with our latest code pushed to it. So here is our docker file.

```
version: '2'
services:
  web:
    image: web:latest
    ports:
      - "8000:8000"
    mem_limit: 7984000
    links:
      - start_dependencies
    depends_on:
      - start_dependencies
  seed-web:
    image: web:seed-latest
    links:
      - start_dependencies
    depends_on:
      - start_dependencies
  db-web:
    # below example references the latest mongo image on docker hub
    image: mongo:3.4.2
  start_dependencies:
    image: web/wait-for-dependencies
    links:
      - db-web
    depends_on:
      - db-web
    command: db-web:27017
```

Bringing up the service with this command:

`ecs-cli compose service up --deployment-min-healthy-percent 0 && success=1`

But after starting multiple tasks the following output is displayed:

```
WARN[0000] Skipping unsupported YAML option...           option name=networks
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=db-web
WARN[0000] Skipping unsupported YAML option for service...  option name="depends_on" service name=seed-web
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=seed-web
WARN[0000] Skipping unsupported YAML option for service...  option name="depends_on" service name="start_dependencies"
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name="start_dependencies"
WARN[0000] Skipping unsupported YAML option for service...  option name="depends_on" service name=web
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=web
INFO[0001] Using ECS task definition                     TaskDefinition="web:23"
INFO[0001] Created an ECS service                        deployment-min-healthy-percent=0 service=web taskDefinition="web:23"
INFO[0001] Updated ECS service successfully              deployment-min-healthy-percent=0 desiredCount=1 serviceName=web
INFO[0016] (service web) has started 1 tasks: (task 89ceda24-dd92-42ab-b290-ec5aa1837206).  timestamp=2018-02-04 09:49:06 +0000 UTC
INFO[0062] (service web) has started 1 tasks: (task 508924a3-1718-4564-8fb8-b95b846597cf).  timestamp=2018-02-04 09:49:49 +0000 UTC
INFO[0137] (service web) has started 1 tasks: (task 56c2f95b-d461-4e1f-ac66-8d36feae39b2).  timestamp=2018-02-04 09:51:07 +0000 UTC
INFO[0182] (service web) has started 1 tasks: (task 581d9318-212a-4e2d-b439-66378918e715).  timestamp=2018-02-04 09:51:51 +0000 UTC
INFO[0228] (service web) has started 1 tasks: (task 8db1f7a0-8217-4f85-ad41-08c4d430cd35).  timestamp=2018-02-04 09:52:34 +0000 UTC
INFO[0273] (service web) has started 1 tasks: (task 824f4aba-10b1-4197-95d0-afe59e3ced41).  timestamp=2018-02-04 09:53:19 +0000 UTC
INFO[0303] (service web) has started 1 tasks: (task cf532423-9618-4a85-aeb4-18de19962bcc).  timestamp=2018-02-04 09:54:04 +0000 UTC
FATA[0303] Deployment has not completed: Running count has not changed for 5.00 minutes
```

I tried setting up cloud watch logging but I got no reports. How do I get the deployment to complete? We are deploying a Node+Express, MongoDB app.

**Comment by SoManyHs** [comment]
Hi @migdall,

Sorry to hear you're having trouble with deployment. One quick observation is that you have the `depends_on` field in your docker compose file, which is not currently supported by ECS. 

Could you possibly include the output of running` aws ecs describe-task-definition --task-definition web:23` (assuming that version of the task definition still exists from running the `compose service up` command)?

**Comment by migdall** [comment]
Thank you for the response. Here is the output of running `aws ecs describe-task-definition --task-definition web:23`  

`An error occurred (ClientException) when calling the DescribeTaskDefinition operation: Unable to describe task definition.`

**Comment by SoManyHs** [comment]
@migdall Does this task definition (e.g. `web:23`) still exist? You can check by running `aws ecs list-task-definitions` for the cluster in question. Alternatively, if the task definition generated from your original command (`ecs-cli compose service up --deployment-min-healthy-percent 0` was cleaned up, you could try running it again, look for the line in the output that tells you the task definition ID (e.g `INFO[0001] Using ECS task definition                     TaskDefinition="web:23"`, in your previous example), and call `aws ecs describe-task-definition` again with whatever the TaskDefinition id is.

**Comment by migdall** [comment]
The task doesn't exist because it fails to stay online. Should I run the test while compose service up is running?

**Comment by SoManyHs** [comment]
Hi @migdall,
I was referring to the [task definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/example_task_definitions.html), not the task itself.  You should also be able to view all your task definitions from the AWS Console, under the "Task Definitions" sidebar item. Are you able to see it there?

**Comment by migdall** [comment]
Yes I can see that task. But all of those task instances are stopped.

**Comment by SoManyHs** [comment]
Hi @migdall,

What is the output when you run `aws ecs list-task-definitions` for the region you are trying to run your task in? Your **Task Definition** should still exist regardless of whether an instance of the **task** that uses that Task Definition is stopped, unless it (the task definition) was deleted after you tried launching the task.

It maybe helpful to check why the task itself was stopped. Common errors for stopped tasks are documented [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/stopped-task-errors.html).

It is also possible, as noted in the [other issue](https://github.com/aws/amazon-ecs-cli/issues/427#issuecomment-366689235) you reported, that your service deployment simply takes longer than the default time. You could try setting the `--timeout` flag to something longer than 5 minutes (which is the default time); alternatively you could specify `--timeout 0` and the CLI will return without checking for success. Regardless, you should still be able to check to see if your service deployment succeeded either through the AWS console or by running `aws ecs list-services`.

Let me know if any of those work for you!

**Comment by frankgibbs** [comment]
I think I found a possible issue. Notice that when you start the stack locally the image dadarek/wait-for-dependencies exits once the other images start. I'm not 100% sure exactly what dadarek/wait-for-dependencies does, but my assumption is that it's supposed to exit. The error in ECS is "STOPPED (Essential container in task exited)". I also noticed that dadarek/wait-for-dependencies in the task definition is tagged as "Essential". Again I don't know exactly what dadarek/wait-for-dependencies does and if it's required in "production" ECS deployment, but I would recommend changing "Essential" to false for this image in the task definition.
